text/0758-default-component-manager.md

```diff
@@ -0,0 +1,138 @@
+---
+Stage: Accepted
+Start Date: 2021-05-17
+Release Date: Unreleased
+Release Versions:
+  ember-source: vX.Y.Z
+  ember-data: vX.Y.Z
+Relevant Team(s): Ember.js, Learning
+RFC PR: https://github.com/emberjs/rfcs/pull/758
+---
+
+# Default Component Manager
+
+## Summary
+
+Anything that can be in a template has its lifecycle managed by the manager pattern.
+Today, we have known managers for `@glimmer/component`, `@ember/helper`, etc.
+But what happens when the VM encounters an object for which there is no manager?,
+such as a plain function? This RFC explores and proposes a default behavior for
+those unknown scenarios when it comes to _components_.
+
+## Motivation
+
+The addon, [ember-could-get-used-to-this](https://github.com/pzuraq/ember-could-get-used-to-this)
+demonstrated that it's possible to use plain functions for helpers and modifiers.
+Components on the other hand, have had no such "plain" / "vanilla" support / adoption from addons.
+Maybe it's worth exploring what default behavior components should have.
+
+This could reduce the number of dependencies folks need to worry about when building components.
+Using native classes eliminates the reliance on a "framework thing" and further
+lessens the appearance of "magic through object-oriented inheritance".
+
+Additionally, a goal of baseclass-less-classes as components is to have 0 potential API
+conflicts with userspace.
```

Comment:
In general, it's rare to want to name something as abstract as `Component`, but in the few cases where it *does* make sense, you can simply take advantage of the fact that ES Modules allow renaming imports (*especially* easy for default exports):

```js
import GlimmerComponent from '@glimmer/component';

export default class Component extends Component {
  // ...
}
```

Accordingly, I don't think this is actually a valuable goal. Names are actually useful and good things to provide, and they provide us great hooks for teaching, for docs, etc.!